tests/TestHarness/Node.py

```diff
@@ -47,21 +43,38 @@ def __init__(self, host, port, nodeId, pid=None, cmd=None, walletMgr=None, nodeo
         self.missingTransaction=False
         self.popenProc=None           # initial process is started by launcher, this will only be set on relaunch
         self.lastTrackedTransactionId=None
-        self.nodeosVers=nodeosVers
-        if self.nodeosVers=="v2":
+        self.pollVersion()
+
+    def pollVersion(self):
+        info = self.getInfo(silentErrors=True)
+        timeout = datetime.now() + timedelta(seconds=20)
+        while info == None:
+            time.sleep(0.5)
+            info = self.getInfo(silentErrors=True)
+            if datetime.now() > timeout:
+                break
+        else:
+            self.configureVersion(info)
+
+    def configureVersion(self, info):
+        self.nodeosVers=info["server_version_string"]
```

Comment:
We should probably have some exceptional or graceful handling of the timeout scenario so that either we have a way to handle in `info` is still `None` and just default to the `else` in `configureVersion` or else raise an exception from `pollVersion` instead of waiting to try to access `info` in `configureVersion`.

Reply:
Added in [27d1430a](https://github.com/AntelopeIO/leap/pull/538/commits/27d1430af550ff6c3d66672baf7d4917368a30e9).